sunflower-land/sunflower-land
# [FEAT] Spring Expansions

# Description

This PR includes the new logic for Spring expansions. Includes:

- Discovering new resources
- Discovering old resources (if a player is under the limit)
- Refunding previous expansions (after each expansion)

**Excludes**

- Spring Biome
- Bumpkin Level Blocking

## How to test?

For a full functional test it is recommended to start a brand new farm and airdrop a lot of resources to expand.

In the FE and BE, update `expandLand.ts` so expansions can be built instanteously.

```
  game.expansionConstruction = {
    createdAt,
    readyAt: createdAt, // + game.expansionRequirements.seconds * 1000,
  };
  ```

There are 2 cases to test:

1. You start a new farm, reach 9 expansions, then upgrade. When you start expanding on Spring island, all the resources should appear.
2. You start a new farm, reach 14+ expansions, then upgrade. When you start expanding on Spring Island, only new resources (Hives, Flower Beds, Crimstone etc.) should appear + you will find a refund reward on the first few expansions.

**Note:** On launch, players above 9 expansions will not be able to expand more. They will need to upgrade
